### PR TITLE
Add Quorum RPC Calls

### PR DESCRIPTION
**Pull request by mayoreee**
Added the following Quorum RPC calls:

- `quorum list`
- `quorum info`
- `quorum dkgstatus`
- `quorum sign`
- `quorum getrecsig`
- `quorum hasrecsig`
- `quorum isconflicting`
- `quorum memberof`
- `quourum rotationinfo`
- `quorum selectquorum`
- `quorum verify`

### CODE REVIEW

File: json/src/lib.rs [hunk 1]
```diff
@@ -2102,6 +2103,271 @@ pub struct MasternodeStatus {
...
+    pub connected: bool,
+    #[serde_as(as = "DisplayFromStr")]
+    pub address: SocketAddr,
+    pub outbound: bool,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct QuorumConnection {
+    pub llmq_type: String,
```
Comment (QuantumExplorer) [inline json/src/lib.rs:2213]: should be an enum no?

Comment (QuantumExplorer) [reply json/src/lib.rs:2213]: I mean QuorumType which should be an enum

File: json/src/lib.rs [hunk 2]
```diff
@@ -2102,6 +2103,271 @@ pub struct MasternodeStatus {
...
+
+
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct QuorumRotationInfo {
+    pub extra_share: bool,
+    pub quorum_snapshot_at_h_minus_c: QuorumSnapshot,
+    pub quorum_snapshot_at_h_minus_2c: QuorumSnapshot,
+    pub quorum_snapshot_at_h_minus_3c: QuorumSnapshot,
+    pub mn_list_diff_tip: MasternodeListDiff,
```
Comment (QuantumExplorer) [inline json/src/lib.rs:2352]: I think camel case will give something like MnListDiffTip when we need MNListDiffTip

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: json/src/lib.rs (hunk 2 above)
Comment (mayoreee) [reply json/src/lib.rs:2359]: @QuantumExplorer camel case here gives `mnListDiffTip`, which is what's in the returned data as seen in the docs here:

Screenshot 2022-09-08 at 17 51 41

Comment (QuantumExplorer) [reply json/src/lib.rs:2359]: Ah okay

File: json/src/lib.rs [hunk 3]
```diff
@@ -2105,6 +2105,13 @@ pub struct MasternodeStatus {
 
 // --------------------------- Quorum -------------------------------
 
+#[serde(untagged)]
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+pub enum QuorumType{
```
Comment (QuantumExplorer) [inline json/src/lib.rs:2110]: This should be enum on the different quorum types with methods to transform into a u8 and to describe as a String.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: json/src/lib.rs (hunk 3 above)
Comment (mayoreee) [reply json/src/lib.rs:2110]: Done

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**